bundles/org.openhab.binding.livisismarthome/src/main/java/org/openhab/binding/livisismarthome/internal/handler/LivisiDeviceHandler.java

```diff
@@ -0,0 +1,1026 @@
+import org.openhab.binding.livisismarthome.internal.util.Translator;
+import org.openhab.core.library.types.DecimalType;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.OpenClosedType;
+import org.openhab.core.library.types.PercentType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.library.types.StopMoveType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.library.types.UpDownType;
+import org.openhab.core.library.unit.SIUnits;
+import org.openhab.core.library.unit.Units;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.CommonTriggerEvents;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingStatusInfo;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LivisiDeviceHandler} is responsible for handling the {@link DeviceDTO}s and their commands, which are
+ * sent to one of the channels.
+ *
+ * @author Oliver Kuhl - Initial contribution
+ * @author Sven Strohschein - Renamed from Innogy to Livisi
+ */
+@NonNullByDefault
+public class LivisiDeviceHandler extends BaseThingHandler implements DeviceStatusListener {
+
+    private static final int MIN_TEMPERATURE = 6;
+    private static final int MAX_TEMPERATURE = 30;
+    private static final String LONG_PRESS = "LongPress";
+    private static final String SHORT_PRESS = "ShortPress";
+
+    private final Logger logger = LoggerFactory.getLogger(LivisiDeviceHandler.class);
+    private final Object lock = new Object();
+    private final Translator translator;
+
+    private String deviceId = "";
+    private @Nullable LivisiBridgeHandler bridgeHandler;
+
+    /**
+     * Constructs a new {@link LivisiDeviceHandler} for the given {@link Thing}.
+     *
+     * @param thing device thing
+     */
+    public LivisiDeviceHandler(final Thing thing, final Translator translator) {
+        super(thing);
+        this.translator = translator;
+    }
+
+    @Override
+    public void handleCommand(final ChannelUID channelUID, final Command command) {
+        logger.debug("handleCommand called for channel '{}' of type '{}' with command '{}'", channelUID,
+                getThing().getThingTypeUID().getId(), command);
+
+        final Optional<LivisiBridgeHandler> bridgeHandlerOptional = getBridgeHandler();
+        if (bridgeHandlerOptional.isPresent()) {
+            LivisiBridgeHandler bridgeHandler = bridgeHandlerOptional.get();
+            if (ThingStatus.ONLINE.equals(bridgeHandler.getThing().getStatus())) {
```

Comment:
If bridge is offline, I believe framework will not send commands, since thing will then also be offline. It will send commands when thing is UNKNOWN or ONLINE. When UNKNOWN you can either ignore commands or try to get thing online when processing command, depending on circumstances.

Reply:
What should I do? Just remove the check?

Reply:
Hmm, I think you can simply check thing's own status:
`if (getThing().getStatus() == ThingStatus.ONLINE) {`

Reply:
But the bridge (handler) is executing the commands. The code is also only working with the bridge handler, so I would think it is correct to check the bridge status.

Reply:
If bridge is offline, so is the thing. Therefore you don't need to check bridge status, but can simply check thing status.

Reply:
solved...